lib/src/component/error_boundary.dart

```diff
@@ -22,4 +24,130 @@ class _$ErrorBoundaryState extends UiState with ErrorBoundaryStateMixin {}
+  //       of any errors thrown.
+  //
+  //   Solution:
+  //     To prevent unmounting the entire tree when React cannot find an Error Boundary that is able
+  //     to handle the error we wrap an Error Boundary with another Error Boundary (this one!). The
+  //     child Error Boundary will handle errors that are "recoverable", so if an error gets to this
+  //     Error Boundary we know it is "unrecoverable" and can present a fallback.
+  //
+  // -----------------------------------------------------------------------------------------------
+  // Implementation:
+  //
+  // [1] Renders a child Error Boundary that is able to handle Errors thrown outside of the initial
+  //     render cycle, allowing it a chance to "recover".
+  //
+  // [2] If we catch an error in this Error Boundary that indicates that the child Error Boundary was
+  //     unable to handle or recover from the error, so we know that it was "unrecoverable" and we
+  //     haven't had a successful render.
+  //
+  //     [2.1] If we have `props.fallbackUIRenderer` we present that.
+  //
+  //     [2.2] Present nothing instead of the broken children.
+  // ---------------------------------------------- /\ ----------------------------------------------
+
+  ReactErrorInfo _errorInfo;
+  dynamic _error;
+
+  @override
+  Map get defaultProps => (newProps()
+    ..identicalErrorFrequencyTolerance = Duration(seconds: 5)
+    ..loggerName = defaultErrorBoundaryLoggerName
+    ..shouldLogErrors = true
+  );
+
+  @override
+  get initialState => (newState()
+    ..hasError = false
+    ..showFallbackUIOnError = true
+  );
+
+  @override
+  Map getDerivedStateFromError(error) => (newState()
+    ..hasError = true
+    ..showFallbackUIOnError = true
+  );
+
+  @override
+  void componentDidCatch(error, ReactErrorInfo info) {
+    _error = error;
+    _errorInfo = info;
+
+    if (props.onComponentDidCatch != null) {
+      props.onComponentDidCatch(error, info);
+    }
+
+    _logErrorCaughtByErrorBoundary(error, info);
+
+    if (props.onComponentIsUnrecoverable != null) {
+      props.onComponentIsUnrecoverable(error, info);
+    }
+  }
+
+  @override
+  render() {
+    if (state.hasError) { // [2]
+      if (props.fallbackUIRenderer != null) {
+        return props.fallbackUIRenderer(_error, _errorInfo); // [2.1]
+      } else {
+        return (Dom.div()
+          ..key = 'ohnoes'
+          ..addTestId('ErrorBoundary.unrecoverableErrorInnerHtmlContainerNode')
+        )(); // [2.2]
+      }
+    }
+    return (RecoverableErrorBoundary()..addProps(props))(props.children); // [1]
+  }
+
+  @override
+  void componentDidUpdate(Map prevProps, Map prevState, [dynamic snapshot]) {
+    // If the child is different, and the error boundary is currently in an error state,
+    // give the children a chance to mount.
+    if (state.hasError) {
+      final childThatCausedError = typedPropsFactory(prevProps).children.single;
+      if (childThatCausedError != props.children.single) {
```

Comment:
this reset logic was actually taken from the original error boundary, i thought it was a way that once there was an error, that if the children passed into the errorboundary were different, it trys again, if we didnt have this logic i think it wouldnt re-render due to its state error. ill add some tests for this if there arent already.

Reply:
Double checked this logic and tested this manually by updating other props and not the children. With this logic if its the same children (aka the one that caused the error) it will NOT re-render (thats good its what we expect and want), but if the children are different this logic resets the state and allows the new child to be rendered, this is good.

One thing to consider though is that we wrapped around components in wdesk like so:
```
ErrorBoundary()(
  SidePanel()()
);
```
that means that that child probably won't change even if the props to it do, but if we were to instead wrap the error boundary around the contents of these components `render` method and new content passed into them would reset the error boundary... 

@aaronlademann-wf @greglittlefield-wf @sydneyjodon-wk @joebingham-wk  thoughts?

Reply:
tldr; from offline convo:

This is not a concern for this PR.